integration/deploy_test.go

```diff
@@ -3,26 +3,28 @@ package integration_test
 import (
 	"time"
 
+	"code.cloudfoundry.org/cf-operator/integration/environment"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("Deploy", func() {
 	Context("when correctly setup", func() {
-		podName := "diego-pod"
+		podName := "my-manifest-nats-v1-0"
+		stsName := "my-manifest-nats-v1"
 
 		AfterEach(func() {
-			env.WaitForPodsDelete(env.Namespace)
+			Expect(env.WaitForPodsDelete(env.Namespace)).To(Succeed())
 		})
 
 		It("should deploy a pod with 10 seconds for the reconciler context", func() {
 			tearDown, err := env.CreateConfigMap(env.Namespace, env.DefaultBOSHManifestConfigMap("manifest"))
 			Expect(err).NotTo(HaveOccurred())
-			defer tearDown()
+			defer func(tdf environment.TearDownFunc) { Expect(tdf()).To(Succeed()) }(tearDown)
```

Comment:
```suggestion
			defer func(f environment.TearDownFunc) { Expect(f()).To(Succeed()) }(tearDown)
```
I think just `f` would be just fine here.